code/modules/atmospherics/machinery/atmosmachinery.dm

```diff
@@ -130,49 +175,106 @@
 	layer = initial(layer) + piping_layer / 1000
 	return ..()
 
+/**
+ * check if a node can actually exists by connecting to another machine
+ * called on atmosinit()
+ * Arguments:
+ * * obj/machinery/atmospherics/target - the machine we are connecting to
+ * * iteration - the current node we are checking (from 1 to 4)
```

Comment:
How does this work with multiz piping?

Reply:
ok so multiz... what the multiz  adapter does, is to change the pipeline_expansion() to something that checks if above and below there is another adapter and add them as nodes.

Now on_construction() calls atmos_init(), then pipeline_expansion() and then for each node again atmos_init()

pipeline_expansion() returns a list of nodes for that device, atmos_init() then does a can_be_node() check for each node it received, and since we already ensured that the above and below adapters exists and they have a node towards each other, can_be_node() will function the same.

Buuut this is what i think it does, i'm not really sure about this